stable/rails/templates/ingress.yaml

```diff
@@ -1,33 +1,37 @@
 {{ if (and .Values.webserver.enabled .Values.ingress.enabled) }}
+{{- range .Values.ingress.list }}
+---
 apiVersion: extensions/v1beta1
 kind: Ingress
 metadata:
-  name: {{ include "rails.name" . }}-ingress
+  name: {{ include "rails.name" $ }}-ingress-{{ .type }}
   annotations:
-  {{- range $key, $value := .Values.ingress.annotations }}
+    kubernetes.io/ingress.class: {{ .type }}
+  {{- range $key, $value := .annotations }}
     {{ $key }}: {{ $value | quote }}
   {{- end }}
   labels:
-    app: {{ include "rails.name" . }}-ingres
-    app.kubernetes.io/name: {{ include "rails.name" . }}
-    app.kubernetes.io/instance: {{ .Release.Name }}
-    app.kubernetes.io/managed-by: {{ .Release.Service }}
-    app.kubernetes.io/version: {{ include "rails.version" . }}
-    environment: {{ include "rails.environment" . }}
-    product: {{ .Values.product }}
-    helm.sh/chart: {{ include "rails.chart" . }}
+    app: {{ include "rails.name" $ }}-ingress
```

Comment:
Could you please explain this change of `.` to `$`. Is it like a context?

Reply:
This change is due to the use of the loop `range`. We changing the field of view in the middle of the loop, so to access the root context change I made this change
https://helm.sh/docs/chart_template_guide/variables/